aten/src/ATen/native/TypeProperties.cpp

```diff
@@ -57,10 +57,13 @@ static inline ScalarType promote_skip_undefined(ScalarType a, ScalarType b) {
 
 
 static inline ScalarType combine_categories(ScalarType higher, ScalarType lower) {
-  if (isFloatingType(higher)) {
+  if(isComplexType(higher)) {
     return higher;
   }
-  if (higher == ScalarType::Bool || isFloatingType(lower)) {
+  if(!isComplexType(lower) && isFloatingType(higher)) {
+    return higher;
+  }
+  if (higher == ScalarType::Bool || isFloatingType(lower) || isComplexType(lower)) {
```

Comment:
The way the rules work for integral and float -- if the `higher` type is floating and the `lower` type is integral, we don't promote, we just use the floating dtype.

I'd expect complex to work the same way -- if the `higher` type, say, a dimensioned tensor, is complex64, and the `lower` type is a zero-dimensional double, I'd expect the result to ignore the floating dtype and return complex64 not complex128.

concretely, right now on master I see I can:
```
((torch.tensor([1j, 1j, 1j], dtype=torch.complex64) * torch.tensor(1.0,dtype=torch.double))).dtype
```
and get back complex128, which I think is incorrect. The dtype of the zero-dimensional floating point tensor should be ignored. I could be mis-reading this but I'm not sure if we fixed that and don't see a particular test for that case.

Reply:
I just tried and I got torch.complex64. maybe this fixed it:  https://github.com/pytorch/pytorch/pull/33713

Reply:
probably fixed by this unlanded PR (my example was from master). let's just make sure there's a test that covers this case.